app/workers/deleted_articles_worker.rb

```diff
@@ -0,0 +1,42 @@
+# frozen_string_literal: true
+class DeletedArticlesWorker
+  include Sidekiq::Worker
+  sidekiq_options lock: :until_executed
+
+  def perform
```

Comment:
I prefer to keep workers very small, and keep all the business logic in a separate file.